worker/create_batches_worker.go

```diff
@@ -274,15 +274,24 @@ func (b *CreateBatchesWorker) Process(message *workers.Msg) {
 		b.checkErr(&msg.Job, err)
 	}
 
-	// pull from db, send to control and send to kafta
+	// pull from db, send to control and send to kafka
 	b.processIDs(ids, &msg)
 
 	completedParts := b.setAsComplete(msg.Part, &msg.Job)
 
 	if completedParts == msg.TotalParts {
 		ids = b.getSplitedIds(msg.TotalParts, &msg.Job)
+
 		b.processIDs(ids, &msg)
-		msg.Job.TagSuccess(b.Workers.MarathonDB, nameCreateBatches, "finished")
+
+		if msg.Job.TotalUsers == 0 {
+			_, err := b.Workers.MarathonDB.Model(&msg.Job).Set("completed_at = ?", time.Now().UnixNano()).Where("id = ?", msg.Job.ID).Update()
+			b.checkErr(&msg.Job, err)
```

Comment:
WDYT of extracting this part to a function updateCompletedAt for maintaining the pattern in this file?

Reply:
Good idea. Will do it.

Reply:
Done